pkgs/development/compilers/gerbil/build.nix

```diff
@@ -106,6 +106,9 @@ stdenv.mkDerivation rec {
     ./install.sh
     (cd $out/bin ; ln -s ../gerbil/bin/* .)
     runHook postInstall
+  '' + lib.optionalString stdenv.isDarwin ''
+    libgerbil="$(realpath "$out/gerbil/lib/libgerbil.so")"
+    install_name_tool -id "$libgerbil" "$libgerbil"
```

Comment:
Looks like this is what `fixDarwinDylibNames` does?

Reply:
After reviewing it definitely is; is there some reason that this wouldn't already being run? I guess the obvious answer would be that the `fixupOutput` hook isn't being run, but wouldn't this package be broken on Linux too if that were the case?

Reply:
It has to be included in `nativeBuildInputs` (of course, only for darwin) for it to be run, at least. It won't be broken on Linux since ELF binaries don't have these identification names, I think.

Reply:
Thanks, I was misunderstanding that it was including in `fixupOutputHooks` by default on Darwin. I'll alter this change to do that instead.

Reply:
Appears there's a fun caveat: `fixDarwinDylibNames` fails because it attempts to process all files with the pattern `*.so.*`; gerbil outputs the linker flags for `libgerbil.so` as `libgerbil.so.ldd`, which means `fixDarwinDylibNames` attempts and fails to fixup it's soname, causing the build to fail.

Reply:
Oh, in that case, not using `fixDarwinDylibNames` is perfectly fine. The change in #202575 has indeed broken a few packages.